### PR TITLE
PEP 691: Draft PEP for Simple JSON API

### PR DESCRIPTION
**Pull request by di**

### CODE REVIEW

File: pep-9999.rst
Comment (JelleZijlstra) [inline pep-9999.rst:1]: You can pick PEP 691

File: pep-0691.rst
Comment (JelleZijlstra) [inline pep-0691.rst:265]:
```suggestion
  <https://docs.python.org/3/library/hashlib.html>`_ (specifically any that can
```

File: pep-0691.rst
Comment (JelleZijlstra) [inline pep-0691.rst:407]: Some missing word here

File: pep-0691.rst
Comment (JelleZijlstra) [inline pep-0691.rst:8]: Are you sure it should be Informational? It reads like a standards-track PEP.

**Review by JelleZijlstra — COMMENTED** [review]
Congratulations on the PEP submission!

Can you also add a CODEOWNERS entry with @dstufft as the owner for this PEP?

File: pep-0691.rst
Comment (di) [reply pep-0691.rst:8]: This is based on https://peps.python.org/pep-0503/ being informational, but I'm fine to change it if this doesn't make sense anymore.

File: pep-0691.rst
Comment (di) [reply pep-0691.rst:407]:
```suggestion
maintainers, who were the two first potential user for the new API. This is
```

File: pep-0691.rst
Comment (JelleZijlstra) [reply pep-0691.rst:8]: Interesting, I feel like it should not be Informational because you are proposing a new standard that multiple tools will have to implement. An Informational PEP would be appropriate if you're just documenting how this API works in Warehouse.

But ultimately that's a decision for the packaging community, so I won't hold up merging the PEP over it. I can merge it now if you like, or wait for your co-authors or others to chime in.

File: pep-0691.rst
Comment (JelleZijlstra) [inline pep-0691.rst:435]: This renders a bit weirdly: https://pep-previews--2578.org.readthedocs.build/pep-0691/#appendix-1-survey-of-use-cases-to-cover

File: pep-0691.rst
Comment (cooperlees) [reply pep-0691.rst:8]: PEP 503 was written post the Simple HTML already existing so I second @JelleZijlstra here - We should probably change.

Comment (di) [reply pep-0691.rst:8]: I agree, but let's see what the wider community thinks in discussion. Fine to merge now!

File: pep-0691.rst
Comment (di) [reply pep-0691.rst:435]: I'm not sure what's causing this. Is this a limitation of rST, an expected style for PEPs, or something syntactically wrong?

File: pep-0691.rst
Comment (di) [reply pep-0691.rst:8]:
```suggestion
Type: Standards Track
```

File: pep-0691.rst
Comment (JelleZijlstra) [reply pep-0691.rst:435]: I think it's indented by one space too many.

File: pep-0691.rst
Comment (JelleZijlstra) [inline pep-0691.rst:442]:
```suggestion
      - Required metadata used (via Package class - https://github.com/pypa/bandersnatch/blob/main/src/bandersnatch/package.py):

        - metadata["info"]
        - metadata["last_serial"]
        - metadata["releases"]

          - digests
            - URL
```

File: pep-0691.rst
Comment (JelleZijlstra) [reply pep-0691.rst]: I applied the change, will confirm it looks OK in the preview and then merge.

**Review by JelleZijlstra — APPROVED** [review]

File: pep-0691.rst
Comment (dstufft) [reply pep-0691.rst]: There's like a 95% chance I picked informational on PEP 503 because I didn't care very much and that seemed fine and nobody called me out on it.

**Comment by CAM-Gerlach** [comment]
I'm very happy to hear that this new API is moving forward, and looking forward to seeing it implemented.

There are a handful of minor technical/rendering issues that should be corrected (inconsistent header order, obsolete BDFL-Delegate header, inconstant code-block syntax, incorrect syntax highlighting, dummy Discussions-To/Post-History, etc.), which I'm happy to help take care of.

However, there are two more critical defects that should be resolved promptly (and should have been spotted and fixed prior to merge), and block me making such changes:

* The Copyright section [required by PEP 1](https://peps.python.org/pep-0001/#what-belongs-in-a-successful-pep) is missing and should be added by an author of the PEP (such as yourself) with the concurrence of the others.
* While @dstufft is the SC-designated `PEP-Delegate` for PyPI infra, while he _is_ allowed to also simultaneously be the PEP's `Sponsor` (given none of the others are core devs or PEP editors, IIRC), [per PEP 1](https://peps.python.org/pep-0001/#submitting-a-pep) he _cannot_ simultaneously be PEP-Delegate _and_ an author of the PEP (as that would be an obvious conflict of interest). Therefore, to resolve this, I would suggest he remove himself as an author of the PEP and serve as the PEP's sponsor instead (alternatively, he could recuse himself as the PEP-Delegate, or apply to the SC for a special exception if no one objected).

Also, while they are not strictly required and not as much a priority immediately, for the benefit of both reviewers and the community at large, I'd advise at some point soon adding the relevant Suggested Sections from [PEP 1](https://peps.python.org/pep-0001/#what-belongs-in-a-successful-pep), [PEP 12](https://peps.python.org/pep-0012/#suggested-sections) and the template, particularly "Security Implications", "Backwards Compatibility" and/or "Reference Implementation". Also, given the "answers" in the "FAQ" each closely fit the purpose of the standard "Rejected Ideas" section, I'd suggest retilting them accordingly.

As a final sidenote, there's no need to include a non-working dummy link for `Discussions-To`; as [PEP 12 states](https://peps.python.org/pep-0012/#how-to-use-this-template), you should instead simply elide it when first submitting the PEP, and then just add it (and `Post-History`) immediately after.

**Pull request merged**
